### ISSUE TITLE
Sentence Pair Generation Question

### ISSUE THREAD

**Issue by rajshah4** [huggingface/setfit#85]
I am trying to understand SetFit - when I run the default code in the readme - 8 examples, 2 classes - I noticed the training dataset has 640 examples.  That 640 is probably going to include duplicate examples, right?  Can you share a bit more detail on the sentence generation process and the impact of duplicates?

**Comment by lewtun** [comment]
Thanks for the question @rajshah4 ! We should probably explain this in a FAQ section of the README :)

The way sentence pair generation works is as follows (see [code](https://github.com/huggingface/setfit/blob/16f9455f563e9a2d2e3394f447c36d42d6031b1d/src/setfit/modeling.py#L204) for more details):

* Given `[s_1,, s_2, ..., s_n]` sentences and `[l_1, l_2, ...., l_n]` labels
* For each `s_i` sample a "positive" `s_j` where `l_i == l_j` and a "negative" `s_k` where `l_i != l_k`. 
* This produces two triples `(s_i, s_j, l_i)` and `(s_i, s_k, l_i)` that are used for contrastive learning
* If we have `N=8` samples per class, and 2 classes, this means each iteration of the above produces `16 x 2 = 32` learning examples. When we iterate 20 times, we thus get `32 x 20 = 640` training examples

Regarding the impact of duplicates, we didn't do an ablation for this in the paper and it would be interesting to explore how performance compares with deduplication applied to the generated pairs. In general, we found that increasing `num_iterations` showed better results, with `num_iterations=20` the sweet spot in our analysis.

**Comment by rajshah4** [comment]
This works for me, thanks closing this issue
